Settings add words

Paullina26 opened the pull request.

Paullina26 committed:
add new component AddWord and style

Paullina26 committed:
fixed bug

Paullina26 committed:
add component Context => UserSettingsProvider

Paullina26 committed:
cleaning code

Paullina26 committed:
data base fixed and getUserSettings

Paullina26 committed:
git wrok?

zoska91 commented on src/components/Navigation/NavigadionMaterialUI.tsx:
```diff
@@ -0,0 +1,150 @@
+  },
+];
+
+const WrapperMenu = styled.div`
```
te style bym wyciągneła do osobnego pliku, trochę się ten plik rozrósł

zoska91 commented on src/components/Navigation/NavigadionMaterialUI.tsx:
```diff
@@ -0,0 +1,150 @@
+    }
+  }
+
+  .css-i4bv87-MuiSvgIcon-root {
```
te cyferki, które masz w nazwach klas - wydaje mi się, że one nie są stałe. Sprawdz czy te elementy nie mają klas bez tego przedrostka "css-i4bv87", czy ten element nie ma np samej klasy "MuiSvgIcon-root"

zoska91 commented on src/components/Navigation/NavigadionMaterialUI.tsx:
```diff
@@ -0,0 +1,150 @@
+      <SpeedDial
+        sx={{ position: 'absolute', bottom: 16, right: 16 }}
+        ariaLabel='Add'
+        // className={classes.SpeedDial}
```
wywal komentarz

zoska91 commented on src/components/Navigation/NavigadionMaterialUI.tsx:
```diff
@@ -0,0 +1,150 @@
+            tooltipTitle={action.name}
+            tooltipOpen
+            onClick={e => {
+              // e.stopPropagation();
```
i tu

zoska91 commented on src/components/Shared/Form/Select.tsx:
```diff
@@ -0,0 +1,77 @@
+
+interface SelectProps {
+  options: string[];
+  onChange: (value: string) => void;
```
czy label dla option jest taki sam jak jego value? 
np będziesz mieć owoce: 
gruszka, jabłko, brzoskwinia. 
to nie chcesz żeby inaczej były wyświetlane, a co innego szło do bazy?

zoska91 replied:
options: {value: string, label: string}[];

zoska91 commented on src/data/option/category_options.tsx:
```diff
@@ -0,0 +1,10 @@
+  'Clothing',
+  'Animals',
+  'Work',
+];
```
to chyba nie do tej apliakcji :)

zoska91 commented on src/data/option/language_options.tsx:
```diff
@@ -0,0 +1 @@
+export const optionsLanguage = ['English', 'German', 'Polish'];
```
na to nie trzeba nego pliku. Options trzymaj sobie w komponencie tak gdzie je wykorzystasz. 
Ewentualnie jak już chcesz nowy plik to obok danego komponentu. Moim zdaniem bez sensu nowy folder gdzieś na to robić.

zoska91 replied:
[value: 'en', label: 'English', ...]

zoska91 commented on src/pages/User/Settings/AddWord.tsx:
```diff
@@ -1,5 +1,87 @@
+
+  // const handleSelectChangeBaseWord = (value: string) => {
+  //   setSelectedOptionLanguageWord(value);
+  // };
```
to będzie potrzebne?

zoska91 commented on src/pages/User/Settings/SettingsApp.tsx:
```diff
@@ -0,0 +1,60 @@
+  ${font_settings(2.4, 'normal', 600)}
+  margin: 5px auto;
+`;
+const SettingsApp = () => {
```
Czemu to się nazywa Setting app? to jakaś nowa aplikacja? Wystarczy Settings

zoska91 commented on src/pages/User/Settings/SettingsApp.tsx:
```diff
@@ -0,0 +1,60 @@
+const SettingsApp = () => {
+  const {
+    defaultWordLanguage,
+    setDefaultWordLanguage,
```
wydaje mi się, że tego w ogóle w tym pliku nie wykorzystujesz.

zoska91 commented on src/pages/User/Settings/SettingsApp.tsx:
```diff
@@ -0,0 +1,60 @@
+    defaultWordLanguage,
+    setDefaultWordLanguage,
+    defaultWordLanguageTranslate,
+    setDefaultWordLanguageTranslate,
```
i tego też

zoska91 commented on src/pages/User/Settings/SettingsApp.tsx:
```diff
@@ -0,0 +1,60 @@
+  const [selectedTranslateLanguageWord, setSelectedTranslateLanguageWord] = useState(
+    defaultWordLanguageTranslate
+  );
+
```
skoro pobierasz te wartości z contexu i masz w nim stan, a do tego tutaj sobie pobierasz funkcje (te, które napisałam, że nie wykorzystujesz) to nie ma sensu tutaj mieć stanu. 
Po prostu tam gdzie wykorzystujesz funkcje z tych useState'ów wykorzystaj te z contextu

zoska91 commented on src/utils/ContextSettingsUser.tsx:
```diff
@@ -0,0 +1,61 @@
+  const { isLoginUser } = useContext(GlobalContext);
+  const [defaultWordLanguage, setDefaultWordLanguage] = useState('English');
+  const [defaultWordLanguageTranslate, setDefaultWordLanguageTranslate] = useState('Polish');
+  
```
te wartości powinny przyjść z bazy.
Na dole masz funkcje pobierającą settings. i to z niej powinnaś pobrać te dane.
I z tego co pamiętam default Language jest dla nas nie ważny. My potrzebujemy znać tylko Translate language, czyli ten, którego user chce się nauczyć. Nie ma dla nas znaczenia z jakiego startuje.

zoska91 commented on src/utils/ContextSettingsUser.tsx:
```diff
@@ -0,0 +1,61 @@
+    setDefaultWordLanguageTranslate,
+  };
+
+  const downloadSettingUser = async () => {
```
chyba lepiej "getUserSettings"?

zoska91 commented on src/utils/ContextSettingsUser.tsx:
```diff
@@ -0,0 +1,61 @@
+        const data = await response.json();
+        console.log('settings are good', data);
+      } else {
+        console.log('settings are not good');
```
tutaj jak dostaniesz te dane to właśnie sobie ten language wrzuć

Paullina26 committed:
fix loader

Paullina26 committed:
fix Mui styles

Paullina26 committed:
fixed and cleaning

Paullina26 committed:
fixed and cleaning navugation, add Login test data

Paullina26 committed:
change colors and fixed navigation

Paullina26 committed:
Fix animation initialization issue on navigation menu

This commit ensures that animations for navigation links are properly triggered upon menu opening and section switching. Added logic to the useEffect monitoring isOpenMenu state to reset and trigger link animations correctly when the navigation is first opened or a section is changed. This update provides smoother user interactions with the navigation menu and fixes the issue where animations were not playing on initial menu expansion after a page reload.

zoska91 commented on src/components/Navigation/Navigation.tsx:
```diff
@@ -8,66 +8,126 @@ interface NavigationProps {
+    });
+  };
+
+  const sectionNavigation = {
```
wyciągnij to do osobnego pliku

zoska91 replied:
bo się strasznie powiększył
